Mage.Sets/src/mage/cards/p/PolygraphOrb.java

```diff
@@ -0,0 +1,138 @@
+        // Put two of them into your hand and the rest into your graveyard. You lose 2 life.
+        Ability ability = new EntersBattlefieldTriggeredAbility(
+                new LookLibraryAndPickControllerEffect(
+                        4, 2, PutCards.HAND, PutCards.GRAVEYARD));
+        ability.addEffect(new LoseLifeSourceControllerEffect(2));
+        this.addAbility(ability);
+
+        // {2}, {T}, Collect evidence 3: Each opponent loses 3 life unless they discard a card or sacrifice a creature.
+        ability = new SimpleActivatedAbility(new PolygraphOrbEffect(), new GenericManaCost(2));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new CollectEvidenceCost(3));
+        this.addAbility(ability);
+    }
+
+    private PolygraphOrb(final PolygraphOrb card) {
+        super(card);
+    }
+
+    @Override
+    public PolygraphOrb copy() {
+        return new PolygraphOrb(this);
+    }
+}
+
+class PolygraphOrbEffect extends OneShotEffect {
+
+    private static final String DISCARD_CHOICE = "Discard a card";
+    private static final String SACRIFICE_CHOICE = "Sacrifice a creature";
+    private static final String LIFE_LOSS_CHOICE = "Lose 3 life";
+
+    PolygraphOrbEffect() {
+        super(Outcome.Detriment);
+        staticText = "Each opponent loses 3 life unless they discard a card or sacrifice a creature";
+    }
+
+    private PolygraphOrbEffect(final PolygraphOrbEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PolygraphOrbEffect copy() {
+        return new PolygraphOrbEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+
+        for (UUID opponentId : game.getOpponents(controller.getId())) {
+            Player opponent = game.getPlayer(opponentId);
+            if (opponent == null) {
+                continue;
+            }
+
+            Set<String> choiceSet = new HashSet<>();
+            if (opponent.getHand().size() > 0) {
+                choiceSet.add(DISCARD_CHOICE);
+            }
+            if (game.getBattlefield().count(StaticFilters.FILTER_CONTROLLED_CREATURE, opponentId, source, game) > 0) {
+                choiceSet.add(SACRIFICE_CHOICE);
+            }
+            choiceSet.add(LIFE_LOSS_CHOICE);
+
+            String chosen;
+            if (choiceSet.size() > 1) {
+                Choice choice = new ChoiceImpl(true);
+                choice.setChoices(choiceSet);
+                opponent.choose(outcome, choice, game);
+                chosen = choice.getChoice();
+                if (chosen == null) {
+                    chosen = LIFE_LOSS_CHOICE;
+                }
+            } else {
+                chosen = LIFE_LOSS_CHOICE;
+            }
+
+            switch (chosen) {
```

Comment:
AI is not part of that PR. Just for info: current code with multiple searching incompatible with AI. Custom dialogs require special code to hint AI like: if opponent is computer and life > 10 then choose life option. Search code “ai hint” for examples.